src/main/java/tdl/client/runner/RecordingSystem.java

```diff
@@ -64,8 +62,8 @@ static void notifyEvent(String lastFetchedRound, String shortName, IConsoleOut c
         }
     }
 
-    static void deployNotifyEvent(String lastFetchedRound, IConsoleOut consoleOut) {
-        notifyEvent(lastFetchedRound, deployToProduction.getShortName(), consoleOut);
+    static void deployNotifyEvent(boolean recordingRequired, String lastFetchedRound) {
+        notifyEvent(recordingRequired, lastFetchedRound, deployToProduction.getShortName());
```

Comment:
I do not like how `recordingRequired` is being passed around.
I think it is time to turn this into a proper interface and if the `recordingRequired` is false then we inject a mock implementation.